File: src/main/java/com/example/dto/response/account/AccountDetailResponse.java
package com.example.dto.response.account;

import com.example.dto.response.card.CardResponse;
import com.example.dto.response.customer.CustomerResponse;
import com.example.util.enums.AccountType;
import com.example.util.enums.CurrencyType;

import java.math.BigDecimal;
import java.util.Set;

public record AccountDetailResponse(Long id, String accountNumber, String ibanCode, AccountType type,
                                    CurrencyType currency, BigDecimal balance, boolean active,
                                    CustomerResponse customer, Set<CardResponse> cards, String lastModifiedDate,
                                    String createdDate, Long version) {
}

File: frontend/src/app/core/models/response/account/account-detail-response.ts
import {AccountResponse} from './account-response';
import {CardResponse} from '../card/card-response';
import {CustomerResponse} from '../customer/customer-response';

export interface AccountDetailResponse extends AccountResponse {
  customer: CustomerResponse;
  cards: CardResponse[];
  createdDate: string;
  version: number;
}
